common/common.go

```diff
@@ -0,0 +1,37 @@
+package common
+
+const (
+	ProviderListenerAddr = ":8000"
+	ClientListenerAddr   = ":8001"
+)
+
+type Nothing struct{}
+
+func NewNothing() *Nothing {
+	return new(Nothing)
+}
+
+type RequestProviderReply struct {
+	Addr string
+}
+
+type RequestProviderError struct {
+	Msg string
+}
+
+func (e *RequestProviderError) Error() string {
+	return e.Msg
+}
+
```

Comment:
It's just the address, but it's a good idea to add an optional username field (which could be valid when the provider joins) to keep track of how much work that particular provider does.